dotnet/CoreLib/MemoryStorage/AzureCognitiveSearch/AzureCognitiveSearchMemory.cs

```diff
@@ -571,5 +565,24 @@ private static double SimilarityToDistance(double similarity)
         return 1 / (2 - similarity);
     }
 
+    private static string BuildSearchFilter(IList<MemoryFilter> filters)
+    {
+        List<string> conditions = new();
+
+        foreach (var filter in filters)
+        {
+            var filterConditions = filter.GetFilters()
+                .Select((keyValue) =>
+                {
+                    var fieldValue = keyValue.Value?.Replace("'", "''", StringComparison.Ordinal);
+                    return $"tags/any(s: s eq '{keyValue.Key}{Constants.ReservedEqualsSymbol}{fieldValue}')";
+                });
+
+            conditions.Add($"({string.Join(" or ", filterConditions)})");
+        }
+
+        return string.Join(" and ", conditions);
```

Comment:
Should this AND/OR logic be reversed? I feel like the more common use case would be to create a single MemoryFilter from search criteria (ie. `QueryHelpers.ParseQuery(uri.Query)`) which in most cases would expect an AND. As it exists now, OR'ing criteria within the same filter seems counter intuitive.

Reply:
This is actually the case in the web client service so I'm thinking it definitely does need to get switched.

Reply:
All rules within one filter are "must have", e.g. use `AND`

Reply:
Changed.